### PR TITLE
 Add cycle: create an infinite, looping range 

### PR DESCRIPTION
**Pull request by Kijewski**
Rebased on #10.

Inspired by [itertools.cycle()][1].

[1]: https://docs.python.org/3.8/library/itertools.html#itertools.cycle

### CODE REVIEW

**Comment by simonask** [comment]
I have taken the liberty of retargeting this PR as well to #10.

File: README.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ following observations:
   storage, unless used as input for further algorithms.
 - `uniq()`: Reduce the output by consecutive equality with `std::unique()`. Does not allocate
   temporary storage, unless used as input for further algorithms.
+- `cycle()`: Create an infinite range repeating the inputs elements in a loop.
```
Comment (simonask) [inline README.md]: inputs -> input

File: include/rx/ranges.hpp [hunk 2]
```diff
@@ -1957,6 +1957,54 @@ struct reverse {
     }
 };
 
+/// Create an infinite range repeating the inputs elements in a loop.
```
Comment (simonask) [inline include/rx/ranges.hpp]: inputs -> input

**Review by simonask — APPROVED** [review]
Good! Just spelling. :)

**Comment by Kijewski** [comment]
Thank you for reviewing! Corrected.

**Review by Hazurl — APPROVED** [review]

**Pull request merged**